test/integration/model/json.test.js

```diff
@@ -508,4 +508,192 @@ describe(Support.getTestDialectTeaser('Model'), () => {
       });
     });
   }
+
+  if (current.dialect.supports.JSONB) {
+    describe('JSONB', () => {
+      beforeEach(function () {
+        this.Event = this.sequelize.define('Event', {
+          data: {
+            type: DataTypes.JSONB,
+            field: 'event_data',
+            index: true
+          },
+          json: DataTypes.JSONB
+        });
+
+        return this.Event.sync({ force: true });
+      });
+
+      it('should create an instance with JSONB data', function () {
+        return this.Event.create({
+          data: {
+            name: {
+              first: 'Homer',
+              last: 'Simpson'
+            },
+            employment: 'Nuclear Safety Inspector'
+          }
+        }).then(() => {
+          return this.Event.findAll().then(events => {
+            const event = events[0];
+
+            expect(event.get('data')).to.eql({
+              name: {
+                first: 'Homer',
+                last: 'Simpson'
+              },
+              employment: 'Nuclear Safety Inspector'
+            });
+          });
+        });
+      });
+
+      it('should query an instance with JSONB data', function () {
+        return this.Event.create({
+          data: {
+            name: {
+              first: 'Homer',
+              last: 'Simpson'
+            },
+            employment: 'Nuclear Safety Inspector'
+          }
+        }).then(() => {
+          return this.Event.findAll({
+            where: {
+              'data.name.first': 'Homer'
+            }
+          }).then(events => {
+            const event = events[0];
+
+            expect(event.get('data')).to.eql({
+              name: {
+                first: 'Homer',
+                last: 'Simpson'
+              },
+              employment: 'Nuclear Safety Inspector'
+            });
+          });
+        });
+      });
+
+      it('should query an instance with JSONB data and order', function () {
```

Comment:
Move this test to `find` section above https://github.com/mkaufmaner/sequelize/blob/b5d77ecedbe96d47d053cb9f11e10cdf0f84d0ae/test/integration/model/json.test.js#L209